### PR TITLE
Adds random OID generation, patient registration via PIX, random UUID…

### PR DESCRIPTION
**Pull request by rcrichton**
… generation for more parts of the document and changes the document to a minmalIC.

Hey @devcritter could you have a look at this.

### CODE REVIEW

**Comment by hnnesv** [comment]
:+1: Looks good to me (I didn't run them or anything, but it looks alright).

As a next step it would be cool to add a registry stored query transaction test as well, so that the complete flow will be tested: provide and register doc -> query for document in registry -> retrieve document set based on stored query result. But anyway not important for this PR (plus I guess anything we manage to do here is in anyway "bonus marks").

_edit: just noticed something minor afterwards. comment below. apologies. still a :+1:_

File: test-send-retrieve.coffee [hunk 1]
```diff
@@ -127,10 +146,14 @@ do ->
...
+      .replace /#{{randomUUID}}/g, -> uuid.v4()
 
-  provideDoc = process fs.readFileSync('pnr-validAphpSampleFullSections.xml').toString()
+  provideDoc = process fs.readFileSync('pnr-minimalIC.xml').toString()
   retrieveDoc = process fs.readFileSync('retrieveDocumentSetb.xml').toString()
+  adt = process fs.readFileSync('adt_a01.er7').toString()
 
-  runTest 'Provide and Register Document Set.b', provideAndRegister, isPnRSuccessful, conf, provideDoc, provideDoc, (success) ->
+  sendRegistyHL7 adt, (success) ->
     if success
```
Comment (hnnesv) [inline test-send-retrieve.coffee:156]: Maybe just add an `else` condition for this with a console log explaining why it failed

**Comment by rcrichton** [comment]
Hey @devcritter I've made the changes. I know we need to add more use cases, I just ran out of time.

**Comment by hnnesv** [comment]
:+1:

**Pull request merged**